### PR TITLE
Add the status of pod when using "Helm status" command, as the issue #2318

### PR DESCRIPTION
**Pull request by yan234280533**
Add the status of pod when using "Helm status" command, as the #2318

### CODE REVIEW

**Comment by yan234280533** [comment]
@thomastaylor312  Can you help me a litter, I  split the commit. But the CI is failed again, when run test-style.  Thanks billions. I don't know how to find a way to solve this problem, so I have no choice but to look for help of you.  Thanks billions.

**Comment by yan234280533** [comment]
@thomastaylor312  I did an ci test, but it failed again. We not edit any code, but it failed also. It so strange.

**Comment by yan234280533** [comment]
@thomastaylor312 I solve the problem by  run the "go fmt" and then commit the file again.

**Comment by yan234280533** [comment]
@technosophos Hi , Matt . Can you help me to review this feature ? The feature is describe in #2318 , it is to add the ability to show the status of pod. Many thanks.

**Comment by yan234280533** [comment]
@adamreese  Hi, adamreese. Can you do a review of this pull requrest for me ?  If these have some questions, I will modify it as soon as possible as you reviewed. So many thanks for you.

**Comment by yan234280533** [comment]
@adamreese  Hi,Adamreese. Can you do an review of this commit, so many thanks.

**Comment by yan234280533** [comment]
@adamreese  Hi , Adamreese. Can you do an review of this commit for me , So many thanks for you. It so sorry to trouble you.

**Comment by technosophos** [comment]
I know a few of us have looked at this, yet nobody's done a full review. Can someone take a moment to do a review?

File: pkg/kube/client.go [hunk 1]
```diff
@@ -629,3 +646,90 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
...
+	if runtime.IsNotRegisteredError(err) {
+		return objPods, nil
+	}
+	if err != nil {
+		return objPods, err
+	}
+
+	selector, err := getSelectorFromObject(versioned)
+
+	log.Printf("getSelectRelationPod selector: %+v", selector)
```
Comment (thomastaylor312) [inline pkg/kube/client.go]: With these log statements, can you make sure to use `c.Log` (you can see examples of this throughout the other functions)

File: pkg/kube/client.go [hunk 2]
```diff
@@ -629,3 +646,90 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
...
+func (c *Client) getRelationPods(infos []*resource.Info) map[string][]api.Pod {
+	var objPods = make(map[string][]api.Pod)
+
+	for _, value := range infos {
+		objPods, _ = c.getSelectRelationPod(value, objPods)
+	}
+	return objPods
+}
+
+func (c *Client) getSelectRelationPod(info *resource.Info, objPods map[string][]api.Pod) (map[string][]api.Pod, error) {
```
Comment (thomastaylor312) [inline pkg/kube/client.go]: I think a comment for at least this function (and probably the others) would be helpful in following this new logic

File: pkg/kube/client.go [hunk 3]
```diff
@@ -629,3 +646,90 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
...
+	pods, err := client.Core().Pods(info.Namespace).List(metav1.ListOptions{
+		FieldSelector: fields.Everything().String(),
+		LabelSelector: labels.Set(selector).AsSelector().String(),
+	})
+	if err != nil {
+		return objPods, err
+	}
+
+	for _, pod := range pods.Items {
+
```
Comment (thomastaylor312) [inline pkg/kube/client.go]: Nit: There is a lot of extra whitespace in this `for` loop, removing some of it may make it cleaner

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Sorry for the long delay on this. A few changes in the comments. Also, could you please write tests for the new functionality?

**Comment by yan234280533** [comment]
@thomastaylor312  I rewrite the PR in an new branch. I am so sorry to trouble you to help me to review the  PR's code.

**Pull request closed without merging**